### PR TITLE
Use MSWord compatible RTF sequence for background text color

### PR DESCRIPTION
**Pull request by tusharsnx**
The `GenRTF(...)` was using `\highlight` control word for sending background text color in the RTF format during a copy command. This doesn't work correctly, since many applications (E.g. MSWord) don't support full RGB with `\highlight`, and instead uses an approximation of what is received. For example, `rgb(197, 15, 31)` becomes `rgb(255, 0, 255)`. Also, the standard way of using background colors is `\cbN` control word, which isn't supported as per the [RTF Spec 1.9.1](https://msopenspecs.azureedge.net/files/Archive_References/[MSFT-RTF].pdf) in Word.

But it briefly mentioned a workaround at Pg. 23, which seems to work on all the RTF editors I tested:

image

The PR makes the changes to use `\chshdng0\chcbpatN` for the background coloring.

Also did some refactoring to make the implementation concise.

## Validation Steps Performed

Verified that the background is correctly copied on below editors:
- MSWord
- WordPad
- LibreOffice
- Outlook

### CODE REVIEW

**Comment by tusharsnx** [comment]
demo

**Comment by DHowett** [comment]
WOW, great find!

**Review by lhecker — COMMENTED** [review]
Nice!

File: src/buffer/out/textBuffer.cpp [hunk 1]
```diff
@@ -2226,17 +2235,29 @@ std::string TextBuffer::GenRTF(const TextAndColor& rows, const int fontHeightPoi
...
+            }
+            else
+            {
+                colorTableBuilder << "\\red" << static_cast<int>(GetRValue(color))
+                                  << "\\green" << static_cast<int>(GetGValue(color))
+                                  << "\\blue" << static_cast<int>(GetBValue(color))
+                                  << ";";
+                colorMap[color] = colorIdx;
+                return colorIdx++;
+            }
```
[the remark this answers is not in the document]
Comment (tusharsnx) [reply src/buffer/out/textBuffer.cpp:2259]: Better! 😄

I searched the entire cpp reference (yeah, I'm a newbie) to see if there's anything like this 😅 I ignored `emplace` as a solution and didn't even check how it works.

**Review by lhecker — APPROVED** [review]

**Review by DHowett — APPROVED** [review]
_glorious!_

**Pull request merged**